addons/sale_mrp/sale_mrp.py

```diff
@@ -85,3 +110,41 @@ def _action_explode(self, move):
         """
         property_ids = move.procurement_id.sale_line_id.property_ids.ids
         return super(StockMove, self.with_context(property_ids=property_ids))._action_explode(move)
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def _get_anglo_saxon_price_unit(self):
+        price_unit = super(AccountInvoiceLine, self)._get_anglo_saxon_price_unit()
+        # in case of anglo saxon with a product configured as invoiced based on delivery, with perpetual
+        # valuation and real price costing method, we must find the real price for the cost of good sold
+        uom_obj = self.env['product.uom']
+        if self.product_id.invoice_policy == "delivery":
+            for s_line in self.sale_line_ids:
+                # qtys already invoiced
+                qty_done = sum([uom_obj._compute_qty_obj(x.uom_id, x.quantity, x.product_id.uom_id) for x in s_line.invoice_lines if x.invoice_id.state in ('open', 'paid')])
+                quantity = uom_obj._compute_qty_obj(self.uom_id, self.quantity, self.product_id.uom_id)
+                # Put moves in fixed order by date executed
+                moves = self.env['stock.move']
+                for procurement in s_line.procurement_ids:
+                    moves |= procurement.move_ids
+                moves.sorted(lambda x: x.date)
+                # Go through all the moves and do nothing until you get to qty_done
+                # Beyond qty_done we need to calculate the average of the price_unit
+                # on the moves we encounter.
+                bom = s_line.product_id.product_tmpl_id.bom_ids and s_line.product_id.product_tmpl_id.bom_ids[0]
+                if bom.type == 'phantom':
+                    average_price_unit = 0
+                    components = s_line._get_bom_component_qty(bom)
+                    for product_id in components.keys():
+                        factor = components[product_id]['qty']
+                        uom = components[product_id]['uom']
```

Comment:
No need for UoM conversion normally as it can all be done in default product's UoM